File: frontend/src/services/cartService.js
import api from "./api";

const cartService = {

  getCart: async () => {
    const { data } = await api.get("/cart");
    return data.data;
  },

  addToCart: async (productId, quantity = 1) => {
    const { data } = await api.post("/cart", {
      productId,
      quantity,
    });

    return data.data;
  },

  updateQuantity: async (cartItemId, quantity) => {
    const { data } = await api.put(`/cart/${cartItemId}`, {
      quantity,
    });

    return data.data;
  },

  removeItem: async (cartItemId) => {
    const { data } = await api.delete(`/cart/${cartItemId}`);
    return data.data;
  },

  clearCart: async () => {
    const { data } = await api.delete("/cart");
    return data.data;
  },
};

export default cartService;

File: backend/src/main/java/com/fitmart/backend/service/CartService.java
package com.fitmart.backend.service;

import com.fitmart.backend.dto.request.AddToCartRequest;
import com.fitmart.backend.dto.request.UpdateCartRequest;
import com.fitmart.backend.dto.response.CartResponse;

public interface CartService {

    CartResponse addToCart(String email, AddToCartRequest request);

    CartResponse updateQuantity(String email,
                                Long cartItemId,
                                UpdateCartRequest request);

    CartResponse removeItem(String email,
                            Long cartItemId);

    CartResponse clearCart(String email);

    CartResponse getCart(String email);

}